examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -0,0 +1,430 @@
+y = survey.target.values.ravel()
+survey.target.head()
+
+###############################################################################
+# We split the sample in a train and a test dataset.
+# Only the train dataset will be used in the following exploratory analysis.
+# This is a way to emulate a real situation where predictions are performed on
+# an unknown target.
+
+from sklearn.model_selection import train_test_split
+
+X_train, X_test, y_train, y_test = train_test_split(
+    X, y, random_state=42
+)
+
+##############################################################################
+# First, let's get some insights by looking at the variable distributions and
+# at the pairwise relationships between them. Only numerical
+# variables will be used.
+
+train_dataset = X_train.copy()
+train_dataset.insert(0, "WAGE", y_train)
+sns.pairplot(train_dataset, diag_kind='kde')
+
+##############################################################################
+# Looking closely at the WAGE distribution it could be noticed that it has a
+# long tail and we could take its logarithm
+# to simplify our problem and approximate a normal distribution.
+# The WAGE is increasing when EDUCATION is increasing.
+# Also, the EXPERIENCE and AGE are linearly correlated.
+#
+# .. _the-pipeline:
+#
+# The pipeline
+# ............
+#
+# To design our machine-learning pipeline, we manually
+# check the type of data that we are dealing with:
+
+survey.data.info()
+
+#############################################################################
+# As seen previously, the dataset contains columns with different data types
+# and we need to apply a specific preprocessing for each data types.
+# In particular categorical variables cannot be included in linear model if not
+# coded as integers first. In addition, to avoid categorical features to be
+# treated as ordered values, we need to one-hot-encode them.
+# Our pre-processor will
+#
+# - one-hot encode (i.e., generate a column by category) the categorical
+#   columns;
+# - replace by 0 and 1 the categories of binary columns;
+# - as a first approach (we will see after how the normalisation of numerical
+#   values will affect our discussion), keep numerical values as they are.
+
+from sklearn.compose import make_column_transformer
+from sklearn.preprocessing import OneHotEncoder
+from sklearn.preprocessing import OrdinalEncoder
+
+categorical_columns = ['RACE', 'OCCUPATION', 'SECTOR']
+binary_columns = ['MARR', 'UNION', 'SEX', 'SOUTH']
+numerical_columns = ['EDUCATION', 'EXPERIENCE', 'AGE']
+
+preprocessor = make_column_transformer(
+    (OneHotEncoder(), categorical_columns),
+    (OrdinalEncoder(), binary_columns),
+    remainder='passthrough'
+)
+
+##############################################################################
+# To describe the dataset as a linear model we choose to use a ridge regressor
+# and to model the logarithm of the WAGE.
+# We sample the complexity parameter space between 1.e-10 and 1.e10.
```

Comment:
```suggestion
# We sample the regularization parameter space between 1.e-10 and 1.e10.
```
Though regularization controls complexity, I find that the term "regularization parameter" is more often use, hence I would prefer it.